Actions/AnalyzeTests/TestResultAnalyzer.ps1

```diff
@@ -1,107 +1,324 @@
+        if ($totalFailed -gt 0) {
+            $failuresSummaryMD = "<i>$totalFailed failing tests, download test results to see details</i>"
+            $failuresSb.Insert(0,"<details><summary>$failuresSummaryMD</summary>") | Out-Null
+            $failuresSb.Append("</details>") | Out-Null
         }
         else {
-            $failuresSb.Insert(0,"<details><summary><i>$totalFailed failing tests</i></summary>") | Out-Null
+            $failuresSummaryMD = "<i>No test failures</i>"
+            $failuresSb.Append($failuresSummaryMD) | Out-Null
+        }
+    }
+    else {
+        $summarySb.Append("<i>No test results found</i>") | Out-Null
+        $failuresSummaryMD = ''
+    }
+    $summarySb.ToString()
+    $failuresSb.ToString()
+    $failuresSummaryMD
+}
+
+function ReadBcptFile {
+    Param(
+        [string] $path
+    )
+
+    if ((-not $path) -or (-not (Test-Path -Path $path -PathType Leaf))) {
+        return $null
+    }
+
+    # Read BCPT file
+    $bcptResult = Get-Content -Path $path -Encoding UTF8 | ConvertFrom-Json
+    $suites = [ordered]@{}
+    # Sort by bcptCode, codeunitID, operation
+    foreach($measure in $bcptResult) {
+        $bcptCode = $measure.bcptCode
+        $codeunitID = $measure.codeunitID
+        $codeunitName = $measure.codeunitName
+        $operation = $measure.operation
+
+        # Create Suite if it doesn't exist
+        if(-not $suites.Contains($bcptCode)) {
+            $suites."$bcptCode" = [ordered]@{}
         }
-        $failuresSb.Append("</details>") | Out-Null
-        if (($summarySb.Length + $failuresSb.Length) -lt 65000) {
-            $summarySb.Append("\n\n$($failuresSb.ToString())") | Out-Null
+        # Create Codeunit under Suite if it doesn't exist
+        if (-not $suites."$bcptCode".Contains("$codeunitID")) {
+            $suites."$bcptCode"."$codeunitID" = @{
+                "codeunitName" = $codeunitName
+                "operations" = [ordered]@{}
+            }
         }
-        else {
-            $summarySb.Append("\n\n<i>$totalFailed failing tests. Download test results to see all</i>") | Out-Null
+        # Create Operation under Codeunit if it doesn't exist
+        if (-not $suites."$bcptCode"."$codeunitID"."operations".Contains($operation)) {
+            $suites."$bcptCode"."$codeunitID"."operations"."$operation" = @{
+                "measurements" = @()
+            }
+        }
+        # Add measurement to measurements under operation
+        $suites."$bcptCode"."$codeunitID"."operations"."$operation".measurements += @(@{
+            "durationMin" = $measure.durationMin
+            "numberOfSQLStmts" = $measure.numberOfSQLStmts
+        })
+    }
+    $suites
+}
+
+function GetBcptSummaryMD {
+    Param(
+        [string] $path,
+        [string] $baseLinePath = '',
+        [string] $thresholdsPath = '',
+        [int] $skipMeasurements = 0,
```

Comment:
It is functionality, which was added because Bardur said that it might be needed
At this time, we are not using it - it is for skipping the x highest measurements of a test run (to cope with warmup scenarioes.)